app/src/components/market/common/currency_selector/index.tsx

```diff
@@ -80,10 +84,16 @@ export const CurrencySelector: React.FC<Props> = props => {
 
   tokens
     .filter(({ address }) => filters.length === 0 || filters.indexOf(address.toLowerCase()) >= 0)
-    .forEach(({ address, image, symbol }, index) => {
+    .forEach(({ address, balance: tokenBalance, decimals, image, symbol }, index) => {
+      const selected = currency && currency.toLowerCase() === address.toLowerCase()
       currencyDropdownData.push({
-        content: <TokenItem image={image} text={symbol} />,
-        extraContent: currency && currency.toLowerCase() === address.toLowerCase() ? balance : '',
+        content: <TokenItem image={image} key={symbol} text={symbol} />,
+        extraContent: selected ? balance : '',
+        secondaryText: !tokenBalance
+          ? ''
+          : Number(tokenBalance) > 0
+          ? formatBigNumber(new BigNumber(tokenBalance), decimals, 5)
```

Comment:
Do we need to format 5 decimals?
The dropdown width is short and the display breaks if there are actually 5 decimals.
![image](https://user-images.githubusercontent.com/4677914/104593470-5410a600-56a2-11eb-8523-b1726b92f7b3.png)

Reply:
Great catch! This seems to cause issues with larger balances too. I ended up keeping the decimals for consistency but removed the hard width set on the secondary text 👍